pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -141,17 +141,39 @@ type Webhook struct {
 	endpoint          string
 	resources         []string
 	operations        []admiv1.OperationType
-	filter            *containers.Filter
+	filters           map[string]*containers.Filter
 	containerRegistry string
 	pinnedLibraries   []libInfo
 }
 
 // NewWebhook returns a new Webhook
 func NewWebhook() (*Webhook, error) {
-	filter, err := apmSSINamespaceFilter()
+
+	var filters map[string]*containers.Filter = make(map[string]*containers.Filter);
+
+	ssiFilter, err := apmSSINamespaceFilter();
+	if err != nil {
+		return nil, err
+	}
+	filters["apm_config.instrumentation.enabled"] = ssiFilter;
+
+	asmFilter, err := apmProductNamespaceFilter("asm_config.enabled_namespaces", "asm_config.disabled_namespaces");
 	if err != nil {
 		return nil, err
 	}
+	filters["asm_config.enabled"] = asmFilter;
+
+	iastFilter, err := apmProductNamespaceFilter("iast_config.enabled_namespaces", "iast_config.disabled_namespaces");
+	if err != nil {
+		return nil, err
+	}
+	filters["iast_config.enabled"] = iastFilter;
```

Comment:
I don't think it is good to have the absolute config name as a key in the filters map. (same applies to the rest).

Using filters["iast"] should be enough, and would be better to create some constant like `const iast = "iast"` and use it wherever needed; this helps avoid runtime errors in case of typo in the key when used as a plain string.

But despite this:
- It is not good to have all the features in the same webhook in the first place